TestFoundation/TestDateInterval.swift

```diff
@@ -10,10 +10,124 @@
 class TestDateInterval: XCTestCase {
     static var allTests: [(String, (TestDateInterval) -> () throws -> Void)] {
         return [
+            ("test_defaultConstructor", test_defaultConstructor),
+            ("test_startEndConstructor", test_startEndConstructor),
+            ("test_startDurationConstructor", test_startDurationConstructor),
+            ("test_compareDifferentStarts", test_compareDifferentStarts),
+            ("test_compareDifferentDurations", test_compareDifferentDurations),
+            ("test_compareSame", test_compareSame),
+            ("test_comparisonOperators", test_comparisonOperators),
+            ("test_intersects", test_intersects),
+            ("test_intersection", test_intersection),
+            ("test_intersectionZeroDuration", test_intersectionZeroDuration),
+            ("test_intersectionNil", test_intersectionNil),
+            ("test_contains", test_contains),
             ("test_hashing", test_hashing),
         ]
     }
 
+    func test_defaultConstructor() {
```

Comment:
We don't use the `constructor` term in Swift. It is called `initializer`: https://docs.swift.org/swift-book/LanguageGuide/Initialization.html